Chore: Code hygiene - warn-numeric-widen

andy-hf-kwok opened the pull request:
## Which issue does this PR close?

This PR aims to address the `-Wnumeric-widen` warning in the codebase.
As the overall scope of strict-warning compliance is quite large, this change focuses on resolving this specific warning type as a first step. Future PRs will continue addressing other warning incrementally. 

Partially closes #. https://github.com/apache/datafusion-comet/issues/2255

## Rationale for this change

To clean up the code as per static check with the hope to resolve all warning and promote the restrict warning profile for the default CI build to uphold Scala code quality. 

## What changes are included in this PR?

To perform explicit type conversion instead of relying on runtime implicit cast.

## How are these changes tested?

* Code can now compile with `-Wnumeric-widen` option on.
* CI continue to pass

mbutrovich commented on pom.xml:
```diff
@@ -738,14 +738,14 @@ under the License.
-                              <arg>-feature</arg>
-                              <arg>-Xlint:_</arg>
-                              <arg>-Ywarn-dead-code</arg>
+<!--                              <arg>-deprecation</arg>-->
```
Are these intended changes down here?

andy-hf-kwok replied:
Nope, reverted.

comphead commented on spark/src/test/scala/org/apache/spark/sql/CometTestBase.scala:
```diff
@@ -558,15 +558,15 @@ abstract class CometTestBase
           record.add(0, i % 2 == 0)
-          record.add(1, i.toByte)
-          record.add(2, i.toShort)
+          record.add(1, i.toByte.toInt)
```
same, can be probably converted directly

andy-hf-kwok commented on spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:
```diff
@@ -416,15 +416,15 @@ abstract class ParquetReadSuite extends CometTestBase {
             record.add(0, i % 2 == 0)
-            record.add(1, i.toByte)
-            record.add(2, i.toShort)
+            record.add(1, i.toByte.toInt)
```
Actually I think if the test suite is mean to examine the behaviour when `Byte` or `Short` being passed, then we should keep it as it is.
I have reverted some `.toInt` on test suites for this purpose and add a suppression annotation instead. 
@comphead @EmilyMatt

andy-hf-kwok commented:
@comphead @mbutrovich I have removed some explicit conversions and added a suppression instead, as it seems more appropriate.
Would you mind to trigger the CI for this PR? 
Thx!
